forkit/commit.py

```diff
@@ -34,7 +34,10 @@ def _commit_related(instance, memo, stack, **kwargs):
             elif isinstance(value, models.Model):
                 _memoize_commit(value, memo=memo, **kwargs)
 
-            setattr(instance, accessor, value)
+            if hasattr(getattr(instance, accessor), 'set'):
+                getattr(instance, accessor).set(value)
+            else:
+                setattr(instance, accessor, value)
```

Comment:
I'd prefer to write this as:

```python
relation = getattr(instance, accessor)
try:
    relation.set(value)
except AttributeError:
    setattr(instance, accessor, value)
```

It avoids calling `getattr` twice and looking for `set` twice. I think it's also a bit clearer to read.

Reply:
I actually prefer the former because it is explicitly saying which attribute can be in error. We don't want to trap errors coming out of the set method 

That said, I don't feel too strongly about it.

Reply:
> it is explicitly saying which attribute can be in error

Ah, good point. I think I should be able to accommodate both requirements. 
:)